tests/cmsis_rtos_v1/src/mutex.c

```diff
@@ -12,29 +12,79 @@
 
 osMutexDef(Mutex_1);
 osMutexDef(Mutex_2);
+osMutexDef(Mutex_multi);
 
 void test_mutex(void)
 {
-	osMutexId mutex_id;
+	osMutexId mutex_id = 0;
 	osStatus status;
+	osMutexId mutex_ids[CONFIG_CMSIS_MUTEX_MAX_COUNT+1];
+	int mtx_cnt = 0;
+	int max_mtx_cnt = CONFIG_CMSIS_MUTEX_MAX_COUNT;
+
+	/* Try deleting invalid mutex object */
+	status = osMutexDelete(mutex_id);
+	zassert_true(status == osErrorParameter,
+		     "Invalid Mutex deleted unexpectedly!");
 
 	mutex_id = osMutexCreate(osMutex(Mutex_1));
 	zassert_true(mutex_id != NULL, "Mutex1 creation failed");
 
+	/* Try to release mutex without obtaining it */
+	status = osMutexRelease(mutex_id);
+	zassert_true(status == osErrorResource, "Mutex released unexpectedly!");
+
+	status = osMutexWait(mutex_id, 0);
+	zassert_true(status == osOK, "Mutex wait failure");
+
+	/* Try to acquire an already acquired mutex */
 	status = osMutexWait(mutex_id, 0);
 	zassert_true(status == osOK, "Mutex wait failure");
 
 	status = osMutexRelease(mutex_id);
 	zassert_true(status == osOK, "Mutex release failure");
 
+	/* Release mutex again as it was acquired twice */
+	status = osMutexRelease(mutex_id);
+	zassert_true(status == osOK, "Mutex release failure");
+
 	status = osMutexDelete(mutex_id);
 	zassert_true(status == osOK, "Mutex delete failure");
+
+	/* Try mutex creation for more than maximum count */
+	for (mtx_cnt = 0; mtx_cnt < max_mtx_cnt+1; mtx_cnt++) {
+		mutex_ids[mtx_cnt] = osMutexCreate(osMutex(Mutex_multi));
+		if (mtx_cnt == max_mtx_cnt) {
+			zassert_true(mutex_ids[mtx_cnt] == NULL,
+			  "Mutex creation pass unexpectedly after max count");
+			for (mtx_cnt = 0; mtx_cnt < max_mtx_cnt; mtx_cnt++) {
+				status = osMutexWait(mutex_ids[mtx_cnt],
+						     osWaitForever);
+				zassert_true(status == osOK, "Mutex wait fail");
+				status = osMutexRelease(mutex_ids[mtx_cnt]);
+				zassert_true(status == osOK,
+					     "Mutex release fail");
+				osMutexDelete(mutex_ids[mtx_cnt]);
+				zassert_true(status == osOK,
+					     "Mutex delete fail");
+			}
```

Comment:
This for loop can also be put in a separate function.. something like cleanup_max_mutex().